ansible_collections/arista/avd/plugins/plugin_utils/topology_generator_utils.py

```diff
@@ -0,0 +1,314 @@
+                        inner_subgraph.attr(rank='same')
+                        for node in nodes:
+                            inner_subgraph.node(node)
+
+            if "groups" in data and data["groups"]:
+                draw_nested_subgraphs(
+                    data["groups"], rank_dict, subgraph, undefined_rank_nodes)
+
+
+def create_graph_and_set_properties():
+    """
+    Create a graphviz graph object and set node and edge properties
+
+    Returns: Graph object
+    """
+    graph_obj = graphviz.Graph(name='parent', format='svg', filename='topology.gv',
+                               graph_attr={
+                                    'splines': 'line'
+                               },
+                               node_attr={
+                                   'color': 'lightblue',
+                                   'style': 'filled',
+                                   'fontname': 'arial',
+                                   'shape': 'box',
+                                   'center': 'true',
+                                   'image': 'switch.jpg',
+                                   'pin': 'True',
+                                   'height': '1',
+                                   'width': '1',
+                                   'fixedsize': 'true',
+                                   'labelloc': 't',
+                                   'fontsize': ' 10pt',
+                                   'fontcolor': 'yellow',
+                               },
+                               edge_attr={
+                                   'fontname': 'arial',
+                                   'fontsize': '8',
+                                   'minlen': '2',
+                                   'center': 'true',
+                                   'concentrate': 'true',
+                                   'weight': '0',
+                                   'labelfloat': 'false'
+                               }
+                               )
+    graph_obj.attr(rank='same')
+    return graph_obj
+
+def generate_topology(rank_dict, node_neighbour_dict, output_list, undefined_rank_nodes):
+    """
+    Generate topology diagram using graphviz.Graph
+
+    Args:
+       rank_dict: Dictionary of nodes with respective levels
+                  ex- {0: ['0'], 1: ['super-spine1', 'super-spine2']}
+       node_neighbour_dict: Dictionary with nodes and neighbours list
+                            ex- {'0': ['super-spine1', 'super-spine2']}
+       output_list: List of dictionaries with nested "name", "nodes" and "groups" keys
+       undefined_rank_nodes: Nodes without parent
+    Returns:
+        None
+    """
+
+    graph_obj = create_graph_and_set_properties()
+
+    draw_nested_subgraphs(
+        output_list, rank_dict, graph_obj, undefined_rank_nodes)
+
+    for node, neighbours in node_neighbour_dict.items():
+        for neighbour in neighbours:
+            if node != '0':
+                graph_obj.edge(node, neighbour)
```

Comment:
More optimal: (only checking if once per node rather than n time per node where n is the size of neighbours)

```suggestion
    for node, neighbours in node_neighbour_dict.items():
        if node != "0": 
            for neighbour in neighbours:
                graph_obj.edge(node, neighbour)
```